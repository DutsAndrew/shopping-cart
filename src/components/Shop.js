import React, { useState, useEffect } from "react";
import '../style/Shop.css';
import SearchBar from "./SearchBar";

const Shop = props => {

  const [cart, setCart] = useState({});

  // async function getSteamGames() {
  //   const api = 'https://steam2.p.rapidapi.com/search/specials/page/1';
  //   const options = {
  //     method: 'GET',
  //     headers: {
  //       'X-RapidAPI-Key': 'dabbbf5224msheae4d25cbe18976p11eaf7jsn7e8d500a641f',
  //       'X-RapidAPI-Host': 'steam2.p.rapidapi.com'
  //     }
  //   };

  //   try {
  //     const response = await fetch(api, options);
  //     const data = await response.json();
  //     console.log(data);
  //   } catch (error) {
  //     console.log(error);
  //     alert('The server could not find what you were looking for, please try again');
  //   }
  // }

  // getSteamGames();

  const onSubmit = (e) => {
    e.preventDefault();
    const inputValue = document.querySelector('#search-input').value;

    // validation
    if (!inputValue.length) {
      return;
    };


    console.log(inputValue);
  }

  return (
    <div id="shop-container">
      <SearchBar submit={onSubmit} />
    </div>
  );
};

export default Shop;