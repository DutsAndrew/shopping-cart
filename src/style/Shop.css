#shop-container {
  display: flex;
  flex-flow: column nowrap;
  justify-content: flex-start;
  align-items: center;
  width: 100vw;
  height: 85vh;
}

#game-tiles {
  display: flex;
  flex-flow: row wrap;
  align-items: center;
  justify-content: center;
  align-content: center;
  flex: auto;
  margin: 0;
  margin-top: 2%;
  width: 100vw;
  height: auto;
  gap: 2%;
}

.game-card {
  display: flex;
  flex-flow: column nowrap;
  justify-content: center;
  align-items: center;
  align-content: center;
  margin-bottom: 2%;
  width: 25vw;
  height: 30vh;
  background-color: rgb(199, 213, 224, 0.1);
}

.game-card-img {
  width: 15vw;
  height: 10vh;
}

.game-card-title {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 20vw;
  height: 3vh;
  font-size: 1em;
}

.game-card-price {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 20vw;
  height: 2vh;
  font-size: 0.75em;
}

.game-card-add-to-cart {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 10vw;
  height: 3vh;
  margin-top: 4%;
  font-size: 1em;
  background-color: #66c0f4;
  color: #1b2838;
}