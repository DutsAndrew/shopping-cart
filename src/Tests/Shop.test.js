import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Shop from '../components/Shop';

describe('Shop component', () => {
  it('renders shop loading image', () => {
    const { container } = render(<Shop />);
    expect(container).toMatchSnapshot();
  });
});