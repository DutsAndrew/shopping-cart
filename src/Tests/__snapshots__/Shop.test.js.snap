// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shop component renders shop loading image 1`] = `
<div>
  <div
    id="loading-container"
  >
    <p
      id="loading-message"
    >
      Loading!
    </p>
    <div
      id="loading-bar"
    />
  </div>
</div>
`;
